compiler/rustc_hir_analysis/src/hir_wf_check.rs

```diff
@@ -66,9 +66,10 @@ fn diagnostic_hir_wf_check<'tcx>(
         fn visit_ty(&mut self, ty: &'tcx hir::Ty<'tcx>) {
             let infcx = self.tcx.infer_ctxt().build();
             let tcx_ty = self.icx.to_ty(ty).fold_with(&mut EraseAllBoundRegions { tcx: self.tcx });
+            let def_id = self.tcx.hir().local_def_id(self.hir_id);
```

Comment:
Should `HirWfCheck` directly contain a `LocalDefId`?

Reply:
yes

Reply:
Yeah, maybe yes. I skipped this refactoring sometimes to have a PR with minimal change, but if make sense I can change it

Reply:
Ok this was easy 😄